src/main/java/dk/kb/datahandler/oai/OaiResponseFilterDrArchive.java

```diff
@@ -42,8 +47,19 @@ public OaiResponseFilterDrArchive(String datasource, DsStorageClient storage, St
      */
     @Override
     public void addToStorage(OaiResponse response) throws ServiceException {
+        SAXParser saxParser = getSaxParser();
+
         for (OaiRecord oaiRecord: response.getRecords()) {
-            String xml = oaiRecord.getMetadata();
+
+            PreservicaOaiRecordHandler handler = new PreservicaOaiRecordHandler();
+
+            try {
+                InputStream inputXml = new ByteArrayInputStream(oaiRecord.getMetadata().getBytes(StandardCharsets.UTF_8));
+                saxParser.parse(inputXml, handler);
+            } catch (SAXException | IOException e) {
+                throw new InternalServiceException(e);
+            }
+
             String recordId = oaiRecord.getId();
             // Preservica StructuralObjects are ignored as they are only used as folders in the GUI.
             if (recordId.contains("oai:so")){
```

Comment:
This could be moved up to avoid parsing structual records that are skipped anyway. Would be a minor performance improvement :-)

Reply:
Nice catch!